### ISSUE TITLE
nesting ReorderableListView throws exception on chrome 

### ISSUE THREAD

**Issue by akasHSarode** [flutter/flutter#64423]
## Steps to Reproduce
This is the code from main.dart file which should be enough to reproduce the issue. 

<details>
 <summary>Minimal reproducible code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(App());
}

class App extends StatelessWidget {
  final String title = 'Nested ReorderableListView Issue';
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: title,
      home: MyHomePage(title: title),
    );
  }
}

class MyHomePage extends StatelessWidget {
  final String title;
  final List outer = ['one', 'two', 'three'];
  final List inner = ['uno', 'dos', 'tres'];

  MyHomePage({this.title});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text(title)),
      body: Center(
        child: Container(
          // main list containing nested reorderable lists
          child: ReorderableListView(
            // OuterListItem contains nested ReorderableListView
            children: outer.map<Widget>((oe) => OuterListItem(
              key: ValueKey(oe),
              children: inner.map((ie) => InnerListItem(
                key: ValueKey('$oe.$ie'),
                text: ie  
              )).toList(),
            )).toList(),
            onReorder: (cIndex, nIndex) {}
          ),
          
          height: 300,
          width: 400,
          padding: EdgeInsets.all(10),
          decoration: BoxDecoration(border: Border.all(width: 5)),
        )
      ),
    );
  }
}

class OuterListItem extends StatelessWidget {
  final Key key; 
  final List<Widget> children;

  const OuterListItem({this.key, this.children}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Container(
      key: key,
      child: ReorderableListView(
        children: children,
        scrollDirection: Axis.horizontal,
        onReorder: (cIndex, nIndex) {}
      ),
      decoration: BoxDecoration(
        border: Border.all(
          color: Colors.pinkAccent,
          width: 3,
        ),
      ),
      margin: EdgeInsets.only(bottom: 10),
      height: 100,
    );
  }
}

class InnerListItem extends StatelessWidget {
  final Key key;
  final String text;

  const InnerListItem({this.key, this.text}): super(key: key);

  @override
  Widget build(BuildContext context) {
    return Card(
      key: key,
      child: Container(
        child: Center(
          child: Text(text),
        ),
        height: 100,
        width: 100,
      ),
      elevation: 2,
      color: Colors.yellow[300],
    );
  }
}
```
</details>

I'm running the code using "flutter run -d chrome"
When I try to reorder any of the outer or inner nested list item, 
animation breaks and following exception is thrown. 

<details>
  <summary>Logs</summary>

```
══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
The following assertion was thrown while handling a gesture:
Assertion failed: file:///C:/flutter/packages/flutter/lib/src/widgets/scroll_controller.dart:113:12
_positions.length == 1
"ScrollController attached to multiple scroll views."

When the exception was thrown, this was the stack:
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 214:49  throw_        
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 24:3    assertFailed  
packages/flutter/src/widgets/scroll_controller.dart 113:30                                                                 get position  
packages/flutter/src/widgets/scroll_controller.dart 120:33                                                                 get offset    
packages/flutter/src/material/reorderable_list.dart 333:51                                                                 [_scrollTo]   
packages/flutter/src/material/reorderable_list.dart 552:11                                                                 <fn>
packages/flutter/src/widgets/drag_target.dart 553:76                                                                       didEnter      
packages/flutter/src/widgets/drag_target.dart 694:22                                                                       <fn>
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/js_array.dart 260:15            firstWhere]   
packages/flutter/src/widgets/drag_target.dart 691:50                                                                       updateDrag    
packages/flutter/src/widgets/drag_target.dart 627:5                                                                        new
packages/flutter/src/widgets/drag_target.dart 388:35                                                                       [_startDrag]  
packages/flutter/src/widgets/drag_target.dart 326:36                                                                       <fn>
packages/flutter/src/gestures/multidrag.dart 272:59                                                                        <fn>
packages/flutter/src/gestures/recognizer.dart 184:24                                                                       invokeCallback
packages/flutter/src/gestures/multidrag.dart 272:14                                                                        [_startDrag]       
packages/flutter/src/gestures/multidrag.dart 262:48                                                                        <fn>
packages/flutter/src/gestures/multidrag.dart 491:14                                                                        accepted
packages/flutter/src/gestures/multidrag.dart 262:10                                                                        acceptGesture      
packages/flutter/src/gestures/arena.dart 265:11                                                                            [_resolveInFavorOf]packages/flutter/src/gestures/arena.dart 224:9                                                                             [_resolve]
packages/flutter/src/gestures/arena.dart 54:12                                                                             resolve
packages/flutter/src/gestures/multidrag.dart 64:17                                                                         resolve
packages/flutter/src/gestures/multidrag.dart 477:7                                                                         [_delayPassed]     
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/isolate_helper.dart 48:19       internalCallback   

Handler: "onStart"
Recognizer:
  DelayedMultiDragGestureRecognizer#b3359
═══════════════════════════════════════════════════════════════════════════════
```

```
[  +86 ms] executing: [C:\flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +40 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 916c3ac648aa0498a70f32b5fc4f6c51447628e3        
[        ] executing: [C:\flutter/] git tag --contains HEAD
[ +152 ms] Exit code 0 from: git tag --contains HEAD
[   +1 ms] 1.20.0
           1.20.0-7.4.pre
           1.20.1
[   +7 ms] executing: [C:\flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +21 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/beta
[        ] executing: [C:\flutter/] git ls-remote --get-url origin
[  +19 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +47 ms] executing: [C:\flutter/] git rev-parse --abbrev-ref HEAD
[  +21 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] beta
[  +33 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.  
[   +3 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.       
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.     
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.     
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.    
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.    
[  +13 ms] executing: C:\android\sdk\platform-tools\adb.exe devices -l
[  +28 ms] List of devices attached
[   +4 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.        
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.  
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.       
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.     
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.     
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.    
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.    
[ +131 ms] Generating
C:\Users\akasH\Desktop\flutterIssues\NestedReorderableListViewIssue\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java
[  +66 ms] Launching lib\main.dart on Chrome in debug mode...
[  +35 ms] Updating assets
[  +57 ms] Syncing files to device Chrome...
[  +25 ms] Generating
C:\Users\akasH\Desktop\flutterIssues\NestedReorderableListViewIssue\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java
[  +44 ms] <- reset
[   +5 ms] C:\flutter\bin\cache\dart-sdk\bin\dart.exe --disable-dart-dev
C:\flutter\bin\cache\artifacts\engine\windows-x64\frontend_server.dart.snapshot --sdk-root C:\flutter\bin\cache\flutter_web_sdk/ --incremental--target=dartdevc --debugger-module-names -Ddart.developer.causal_async_stacks=true --output-dill
C:\Users\akasH\AppData\Local\Temp\flutter_tools.c9708663\flutter_tool.91cfb19e\app.dill --libraries-spec
file:///C:/flutter/bin/cache/flutter_web_sdk/libraries.json --packages .packages -Ddart.vm.profile=false -Ddart.vm.product=false
--bytecode-options=source-positions,local-var-info,debugger-stops,instance-field-initializers,keep-unreachable-code,avoid-closure-call-instructions --enable-asserts --track-widget-creation --filesystem-root
C:\Users\akasH\AppData\Local\Temp\flutter_tools.c9708663\flutter_tools.d2f199ab --filesystem-scheme org-dartlang-app --initialize-from-dill   
build\cache.dill.track.dill --platform file:///C:/flutter/bin/cache/flutter_web_sdk/kernel/flutter_ddc_sdk.dill
[   +5 ms] <- compile org-dartlang-app:/web_entrypoint.dart
[+5835 ms] Syncing files to device Chrome... (completed in 5,910ms, longer than expected)
[        ] Synced 23.0MB.
[        ] <- accept
[        ] Caching compiled dill
[ +128 ms] [CHROME]:
[   +2 ms] [CHROME]:DevTools listening on ws://127.0.0.1:52373/devtools/browser/4d53cb6b-f272-4797-a965-b12e548d858a
[+3697 ms] Debug service listening on ws://127.0.0.1:52395/4GaNnFDPfCY=

[ +171 ms] Debug service listening on ws://127.0.0.1:52395/4GaNnFDPfCY=
[   +1 ms] Warning: Flutter's support for web development is not stable yet and hasn't
[   +3 ms] been thoroughly tested in production environments.
[  +25 ms] For more information see https://flutter.dev/web
[        ] �🔥  To hot restart changes while running, press "r" or "R".
[   +3 ms] For a more detailed help message, press "h". To quit, press "q".
```

```
Analyzing NestedReorderableListViewIssue...
No issues found! (ran in 1.9s)
```

```
[√] Flutter (Channel beta, 1.20.0, on Microsoft Windows [Version 10.0.18362.1016], locale en-IN)
    • Flutter version 1.20.0 at C:\flutter
    • Framework revision 916c3ac648 (3 weeks ago), 2020-08-01 09:01:12 -0700
    • Engine revision d6ee1499c2
    • Dart version 2.9.0 (build 2.9.0-21.10.beta)


[!] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at C:\android\sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = C:\android\sdk
    • ANDROID_SDK_ROOT = C:\android\sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[!] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.48.1)
    • VS Code at C:\Users\User\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.13.2

[√] Connected device (3 available)
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome 84.0.4147.135
    • Edge (web)       • edge       • web-javascript • Microsoft Edge 84.0.522.63

! Doctor found issues in 2 categories.
```

</details>

**Comment by markusaksli-nc** [comment]
This error will occur on all platforms when you nest scrollables with ReorderableListView. You can try the advice mentioned in https://github.com/flutter/flutter/issues/62855#issuecomment-669132093 to get around this. 

I'm leaving this open since the error message and documentation for this can be improved.

**Comment by akasHSarode** [comment]
I previously got it to work by changing the source code of ReorderableListView to use a new ScrollController() instead of looking for one in the context. They have now exposed the scrollController property with this commit https://github.com/flutter/flutter/commit/fd1062debd531ae1815e0b2deb647e4e260af7bf, it doesn't throw any exception now but when trying to reorder nested list other than the first one, it behaves abnormally.

**Comment by maheshmnj** [comment]
@akasHSarode I verified the issue on the latest master channel and the issue does not reproduce anymore. 

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const App());
}

class App extends StatelessWidget {
  final String title = 'Nested ReorderableListView Issue';

  const App({super.key});
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: title,
      home: MyHomePage(title: title),
    );
  }
}

class MyHomePage extends StatelessWidget {
  final String title;
  final List outer = ['one', 'two', 'three'];
  final List inner = ['uno', 'dos', 'tres'];

  MyHomePage({super.key, required this.title});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text(title)),
      body: Center(
          child: Container(
        // main list containing nested reorderable lists
        height: 300,
        width: 400,
        padding: const EdgeInsets.all(10),
        decoration: BoxDecoration(border: Border.all(width: 5)),
        // main list containing nested reorderable lists
        child: ReorderableListView(
            // OuterListItem contains nested ReorderableListView
            children: outer
                .map<Widget>((oe) => OuterListItem(
                      key: ValueKey(oe),
                      children: inner
                          .map((ie) =>
                              InnerListItem(key: ValueKey('$oe.$ie'), text: ie))
                          .toList(),
                    ))
                .toList(),
            onReorder: (cIndex, nIndex) {}),
      )),
    );
  }
}

class OuterListItem extends StatelessWidget {
  @override
  final Key key;
  final List<Widget> children;

  const OuterListItem({required this.key, required this.children})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Container(
      key: key,
      decoration: BoxDecoration(
        border: Border.all(
          color: Colors.pinkAccent,
          width: 3,
        ),
      ),
      margin: const EdgeInsets.only(bottom: 10),
      height: 100,
      child: ReorderableListView(
          scrollDirection: Axis.horizontal,
          onReorder: (cIndex, nIndex) {},
          children: children),
    );
  }
}

class InnerListItem extends StatelessWidget {
  @override
  final Key key;
  final String text;

  const InnerListItem({required this.key, required this.text})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Card(
      key: key,
      elevation: 2,
      color: Colors.yellow[300],
      child: Container(
        child: Center(
          child: Text(text),
        ),
        height: 100,
        width: 100,
      ),
    );
  }
}

```
</details> 

<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel master, 3.7.0-3.0.pre.33, on macOS 12.6 21G115 darwin-arm64, locale en-IN)
    • Flutter version 3.7.0-3.0.pre.33 on channel master 
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5201856805 (38 minutes ago), 2022-12-05 18:27:21 -0800
    • Engine revision a309d239c4
    • Dart version 2.19.0 (build 2.19.0-463.0.dev)
    • DevTools version 2.20.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly
      to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.70.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.55.20221129

[✓] Connected device (3 available)
    • iPhone 12 Pro (mobile) • 026D5789-9E78-4AD5-B1B2-3F8D4E7F65E4 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.6 21G115 darwin-arm64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 108.0.5359.94

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

### logs

```console
mahesh@Maheshs-MacBook-Air-M1 flutter_issue % flutter run 
Could not load custom device from config index 0: Expected enabled to be a boolean.
Running "flutter pub get" in flutter_issue...                    1,159ms
mahesh@Maheshs-MacBook-Air-M1 flutter_issue % flutterm clean
Cleaning Xcode workspace...                                      2,581ms
Cleaning Xcode workspace...                                      2,225ms
Deleting build...                                                   75ms
Deleting .dart_tool...                                              15ms
Deleting Generated.xcconfig...                                       0ms
Deleting flutter_export_environment.sh...                            0ms
Deleting Flutter.podspec...                                          0ms
Deleting ephemeral...                                                0ms
Deleting .flutter-plugins-dependencies...                            0ms
Deleting .flutter-plugins...                                         0ms
mahesh@Maheshs-MacBook-Air-M1 flutter_issue % flutterm run
Could not load custom device from config index 0: Expected enabled to be a boolean.
Multiple devices found:
sdk gphone arm64 (mobile) • emulator-5554                        • android-arm64 • Android 11 (API 30) (emulator)
iPhone 12 Pro (mobile)    • 026D5789-9E78-4AD5-B1B2-3F8D4E7F65E4 • ios           • com.apple.CoreSimulator.SimRuntime.iOS-14-5
(simulator)
[1]: sdk gphone arm64 (emulator-5554)
[2]: iPhone 12 Pro (026D5789-9E78-4AD5-B1B2-3F8D4E7F65E4)
Please choose one (To quit, press "q/Q"): 1
Using hardware rendering with device sdk gphone arm64. If you notice graphics artifacts, consider enabling software rendering with
"--enable-software-rendering".
Launching lib/main.dart on sdk gphone arm64 in debug mode...
Running Gradle task 'assembleDebug'...                             19.3s
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app-debug.apk...          940ms
Syncing files to device sdk gphone arm64...                        199ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h List all available interactive commands.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).

💪 Running with sound null safety 💪

An Observatory debugger and profiler on sdk gphone arm64 is available at: http://127.0.0.1:54347/FDLqgGW54PM=/
The Flutter DevTools debugger and profiler on sdk gphone arm64 is available at:
http://127.0.0.1:9101?uri=http://127.0.0.1:54347/FDLqgGW54PM=/
I/e.flutter_issu( 4773): Waiting for a blocking GC ProfileSaver
I/e.flutter_issu( 4773): WaitForGcToComplete blocked ProfileSaver on ClassLinker for 13.648ms

Performing hot reload...                                                
Reloaded 1 of 646 libraries in 399ms (compile: 84 ms, reload: 175 ms, reassemble: 48 ms).


══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
mahesh@Maheshs-MacBook-Air-M1 flutter_issue % flutterm run -d chrome
This application is not configured to build on the web.
To add web support to a project, run `flutter create .`.
Launching lib/main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...             12.6s
This app is linked to the debug service: ws://127.0.0.1:56814/1SrBANS0dBA=/ws
Debug service listening on ws://127.0.0.1:56814/1SrBANS0dBA=/ws

💪 Running with sound null safety 💪

🔥  To hot restart changes while running, press "r" or "R".
For a more detailed help message, press "h". To quit, press "q".

An Observatory debugger and profiler on Chrome is available at: http://127.0.0.1:56814/1SrBANS0dBA=
The Flutter DevTools debugger and profiler on Chrome is available at: http://127.0.0.1:9101?uri=http://127.0.0.1:56814/1SrBANS0dBA=

```

Closing as fixed. Incase you still encounter the issue, Please feel free to write in the comments and we will reopen it.

Thank you.
